psa-crypto/src/types/key.rs

```diff
@@ -305,6 +305,37 @@ impl Attributes {
         get_attributes_res?;
         Ok(attributes?)
     }
+
+    /// Sufficient size for a buffer to export the key, if supported
+    #[cfg(feature = "with-mbed-crypto")]
+    pub fn export_key_output_size(self) -> Result<usize> {
+        match self.key_type {
```

Comment:
The specification says that this macro allows for `key_type`:

> `key_type` A public key or key pair key type.

So I think you can add DH both for key pair and public key as a valid type as well.

Reply:
Is DH the same as DSA? I've been looking at [this](https://github.com/ARMmbed/mbedtls/blob/bd004e862d8479586d7622811d02c65f44544125/include/psa/crypto_sizes.h#L652) to try and make sure its not possible to pass it something invalid.

Reply:
> Is DH the same as DSA?

No, Diffie-Hellman is a type of key exchange, Digital Signature Algorithm is (as its name says) an asymmetric signature algorithm. 

For DH public keys, the spec dictates:
> For Diffie-Hellman key exchange public keys, with key types for which PSA_KEY_TYPE_IS_DH_PUBLIC_KEY is true, the format is the representation of the public key y = g^x mod p as a big-endian byte string. The length of the byte string is the length of the base prime p in bytes.

Which brings me to another realisation - that method there, `export_key_output_size` is improperly named, as it refers only to public keys, not to keys in general.

Reply:
Ah, I see, you're using the spec-provided macros, I think that's alright (although, will those disappear in the future?). But we do have to make sure we document whether this method refers to the full key or just the public part